diff -Nrup lua-5.4.6.orig/src/Makefile lua-5.4.6/src/Makefile
--- lua-5.4.6.orig/src/Makefile	2023-09-03 06:09:08.822274000 +0100
+++ lua-5.4.6/src/Makefile	2023-09-03 06:09:17.989996600 +0100
@@ -7,7 +7,7 @@
 PLAT= guess
 
 CC= gcc -std=gnu99
-CFLAGS= -O2 -Wall -Wextra -DLUA_COMPAT_5_3 $(SYSCFLAGS) $(MYCFLAGS)
+CFLAGS= -O2 -Wall -Wextra $(SYSCFLAGS) $(MYCFLAGS)
 LDFLAGS= $(SYSLDFLAGS) $(MYLDFLAGS)
 LIBS= -lm $(SYSLIBS) $(MYLIBS)
 
diff -Nrup lua-5.4.6.orig/src/luaconf.h lua-5.4.6/src/luaconf.h
--- lua-5.4.6.orig/src/luaconf.h	2023-09-03 06:09:08.824276300 +0100
+++ lua-5.4.6/src/luaconf.h	2023-09-03 06:11:48.925515700 +0100
@@ -202,29 +202,28 @@
 ** In Windows, any exclamation mark ('!') in the path is replaced by the
 ** path of the directory of the executable file of the current process.
 */
-#define LUA_LDIR	"!\\lua\\"
-#define LUA_CDIR	"!\\"
-#define LUA_SHRDIR	"!\\..\\share\\lua\\" LUA_VDIR "\\"
+#define LUA_LDIR	"!\\..\\local\\lib\\lua\\" LUA_VDIR "\\"
+#define LUA_CDIR	"!\\..\\lib\\lua\\" LUA_VDIR "\\"
 
 #if !defined(LUA_PATH_DEFAULT)
 #define LUA_PATH_DEFAULT  \
 		LUA_LDIR"?.lua;"  LUA_LDIR"?\\init.lua;" \
 		LUA_CDIR"?.lua;"  LUA_CDIR"?\\init.lua;" \
-		LUA_SHRDIR"?.lua;" LUA_SHRDIR"?\\init.lua;" \
 		".\\?.lua;" ".\\?\\init.lua"
 #endif
 
+
 #if !defined(LUA_CPATH_DEFAULT)
 #define LUA_CPATH_DEFAULT \
+		"!\\?.dll;" \
 		LUA_CDIR"?.dll;" \
-		LUA_CDIR"..\\lib\\lua\\" LUA_VDIR "\\?.dll;" \
 		LUA_CDIR"loadall.dll;" ".\\?.dll"
 #endif
 
 #else			/* }{ */
 
-#define LUA_ROOT	"/usr/local/"
-#define LUA_LDIR	LUA_ROOT "share/lua/" LUA_VDIR "/"
+#define LUA_ROOT	"/opt/wonderful/"
+#define LUA_LDIR	LUA_ROOT "local/lib/lua/" LUA_VDIR "/"
 #define LUA_CDIR	LUA_ROOT "lib/lua/" LUA_VDIR "/"
 
 #if !defined(LUA_PATH_DEFAULT)
