From 95f148c2222db608a0d72d5c5577d0c71e7fa199 Mon Sep 17 00:00:00 2001
From: Allan McRae <allan@archlinux.org>
Date: Mon, 4 Mar 2024 13:39:37 +1000
Subject: [PATCH] Update website to reflect gitlab usage

Signed-off-by: Allan McRae <allan@archlinux.org>
---
 doc/index.asciidoc | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/doc/index.asciidoc b/doc/index.asciidoc
index 10981a7a9..0bc123aff 100644
--- a/doc/index.asciidoc
+++ b/doc/index.asciidoc
@@ -34,6 +34,8 @@ makepkg/repo-add/pacman toolchain via GnuPG and GPGME.
 Version 5.0 added support for pre/post-transaction hooks and sync database file
 list operations.
 
+Version 6.0 added support for parallel downloads.
+
 Documentation
 -------------
 
@@ -192,7 +194,9 @@ Releases
 
 |======
 
-Source code for all releases is available at
+Source code for releases since moving to gitlab is available at
+link:https://gitlab.archlinux.org/pacman/pacman/-/releases[].
+Source code for historical releases is available at
 link:https://sources.archlinux.org/other/pacman/[]. To install, download the newest
 available source tarball, unpack it in a directory, and run the three magic
 commands:
@@ -240,14 +244,10 @@ link:translation-help.html[translation-help].
 
 Bugs
 ----
-If you find bugs (which is quite likely), please email them to the pacman-dev
-mailing last at mailto:pacman-dev@lists.archlinux.org[] with specific information
-such as your command-line, the nature of the bug, and even the package database
-if it helps.
-
-You can also post a bug to the Arch Linux bug tracker
-link:https://bugs.archlinux.org/index.php?project=3[Flyspray]. Be sure to file
-bugs under the Pacman project.
+If you find bugs (which is quite likely), please report them to the issue
+tracker at link:https://gitlab.archlinux.org/pacman/pacman/-/issues[] with
+specific information such as your command-line, the nature of the bug, and even
+the package database if it helps.
 
 Copyright
 ---------
-- 
GitLab

