diff -rup pacman-6.1.0.orig/lib/libalpm/add.c pacman-6.1.0/lib/libalpm/add.c
--- pacman-6.1.0.orig/lib/libalpm/add.c	2024-05-07 16:34:54.741806813 +0200
+++ pacman-6.1.0/lib/libalpm/add.c	2024-05-07 16:35:29.355140335 +0200
@@ -115,9 +115,7 @@ static int perform_extraction(alpm_handl
 {
 	int ret;
 	struct archive *archive_writer;
-	const int archive_flags = ARCHIVE_EXTRACT_OWNER |
-	                          ARCHIVE_EXTRACT_PERM |
-	                          ARCHIVE_EXTRACT_TIME |
+	const int archive_flags = ARCHIVE_EXTRACT_TIME |
 	                          ARCHIVE_EXTRACT_UNLINK |
 	                          ARCHIVE_EXTRACT_XATTR |
 	                          ARCHIVE_EXTRACT_SECURE_SYMLINKS;
diff -rup pacman-6.1.0.orig/src/pacman/check.c pacman-6.1.0/src/pacman/check.c
--- pacman-6.1.0.orig/src/pacman/check.c	2024-05-07 16:34:54.771806814 +0200
+++ pacman-6.1.0/src/pacman/check.c	2024-05-07 16:35:45.435140426 +0200
@@ -388,8 +388,6 @@ int check_pkg_full(alpm_pkg_t *pkg)
 			continue;
 		}
 
-		file_errors += check_file_permissions(pkgname, filepath, &st, entry);
-
 		if(type == AE_IFLNK) {
 			file_errors += check_file_link(pkgname, filepath, &st, entry);
 		}
diff -rup pacman-6.1.0.orig/src/pacman/util.c pacman-6.1.0/src/pacman/util.c
--- pacman-6.1.0.orig/src/pacman/util.c	2024-05-07 16:34:54.771806814 +0200
+++ pacman-6.1.0/src/pacman/util.c	2024-05-07 16:35:54.948473812 +0200
@@ -122,6 +122,10 @@ int trans_release(void)
 
 int needs_root(void)
 {
+	if (1) {
+		return 0;
+	}
+
 	switch(config->op) {
 		case PM_OP_DATABASE:
 			return !config->op_q_check;
