diff -rup libpng-1.6.34.orig/Makefile.am libpng-1.6.34/Makefile.am
--- libpng-1.6.34.orig/Makefile.am	2018-03-29 10:12:44.884014953 +0200
+++ libpng-1.6.34/Makefile.am	2018-03-29 10:13:09.714899431 +0200
@@ -12,13 +12,13 @@ PNGLIB_BASENAME= libpng@PNGLIB_MAJOR@@PN
 ACLOCAL_AMFLAGS = -I scripts
 
 # test programs - run on make check, make distcheck
-check_PROGRAMS= pngtest pngunknown pngstest pngvalid pngimage pngcp
+check_PROGRAMS= 
 if HAVE_CLOCK_GETTIME
-check_PROGRAMS += timepng
+check_PROGRAMS += 
 endif
 
 # Utilities - installed
-bin_PROGRAMS= pngfix png-fix-itxt
+bin_PROGRAMS= 
 
 # This ensures that pnglibconf.h gets built at the start of 'make all' or
 # 'make check', but it does not add dependencies to the individual programs,
diff -rup libpng-1.6.34.orig/Makefile.in libpng-1.6.34/Makefile.in
--- libpng-1.6.34.orig/Makefile.in	2018-03-29 10:12:44.889014930 +0200
+++ libpng-1.6.34/Makefile.in	2018-03-29 10:13:18.377859132 +0200
@@ -100,11 +100,8 @@ PRE_UNINSTALL = :
 POST_UNINSTALL = :
 build_triplet = @build@
 host_triplet = @host@
-check_PROGRAMS = pngtest$(EXEEXT) pngunknown$(EXEEXT) \
-	pngstest$(EXEEXT) pngvalid$(EXEEXT) pngimage$(EXEEXT) \
-	pngcp$(EXEEXT) $(am__EXEEXT_1)
-@HAVE_CLOCK_GETTIME_TRUE@am__append_1 = timepng
-bin_PROGRAMS = pngfix$(EXEEXT) png-fix-itxt$(EXEEXT)
+check_PROGRAMS = 
+bin_PROGRAMS = 
 @PNG_ARM_NEON_TRUE@am__append_2 = arm/arm_init.c\
 @PNG_ARM_NEON_TRUE@	arm/filter_neon.S arm/filter_neon_intrinsics.c
 
